### DISCUSSION TITLE
inplace rotation is not undefined

### DISCUSSION THREAD

**Discussion by CatInTheRain** [ceres-solver/ceres-solver#1119] [Ideas]
Hi all. I was thinking about the motivation to forbid inplace rotation for the function AngleAxisRotatePoint in rotation.h.
By inspecting the code I see that the `result` variable is always a function of `pt` variable only:
```c
result[i] = pt[i] + ...
```
Sorry for my poor experience but where is the undefined behaviour when `pt` and `result` points to the same address memory location?

**Comment by sergiud** [comment]
I agree. The check could be removed.
